src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern(
+            "dd.MM.yyyy", Locale.ENGLISH);
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int[] salaryForPeriod = new int[names.length];
+        LocalDate beginDate = LocalDate.parse(dateFrom, DATE_TIME_FORMATTER);
+        LocalDate endDate = LocalDate.parse(dateTo, DATE_TIME_FORMATTER);
+        for (int i = 0; i < names.length; i++) {
+            int employerSalary = 0;
+            for (String datum : data) {
+                String[] dataFormatted = datum.split(" ");
+                LocalDate workingDayDate = LocalDate.parse(dataFormatted[0], DATE_TIME_FORMATTER);
+                if (workingDayDate.isAfter(beginDate.minusDays(1))
+                        && workingDayDate.isBefore(endDate.plusDays(1))
+                        && names[i].equals(dataFormatted[1])) {
+                    employerSalary = employerSalary + Integer.parseInt(dataFormatted[2])
+                            * Integer.parseInt(dataFormatted[3]);
+                }
+            }
+            salaryForPeriod[i] = employerSalary;
+        }
+
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo)
+                .append(System.lineSeparator());
+        for (int i = 0; i < names.length; i++) {
+            stringBuilder.append(names[i]).append(" - ").append(salaryForPeriod[i]);
+            if (i < names.length - 1) {
+                stringBuilder.append(System.lineSeparator());
+            }
+        }
```

Comment:
1. `if` is not required if wqe move separator to beginning of line
2. better to split append commands for more easy code reading
3. We can proceed to append right after new StringBuilder()
```suggestion
        StringBuilder stringBuilder = new StringBuilder()
                .append("Report for period ")
                .append(dateFrom)
                .append(" - ")
                .append(dateTo);

        for (int i = 0; i < names.length; i++) {
            stringBuilder
                    .append(System.lineSeparator())
                    .append(names[i])
                    .append(" - ")
                    .append(salaryForPeriod[i]);
        }
```